include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl.h

```diff
@@ -1174,7 +1177,8 @@ __parallel_find_or(oneapi::dpl::__internal::__device_backend_tag, _ExecutionPoli
                             for (auto __old = __found.load(); __comp(__found_local.load(), __old);
```

Comment:
I believe all writes to `__found_local` happened before the barrier.  Can we just read from `__temp_local` directly now without involving atomics?

Reply:
`__found_local` describes the state of find inside one group. So atomic required, I believe.

Reply:
It is the workgroup-local "found" status, but all writes to this variable occur before the intra-workgroup barrier and we are now after the barrier. This should mean that **all** threads in the workgroup are past the barrier, and only reads to this memory remain.  I believe this means we can interact directly with the memory without any danger.  

My guess is that it wont change performance, but it has a chance for an improvement.

Reply:
You should be able to replace all instances of `found_local.load()` with `__temp_local[0]` after the barrier (or copy it into some local variable `found_local_state` for clarity.  Concurrent reads are safe, all writes will be resolved before the barrier.

Reply:
I have checked this variant: there is no significant performance improvement during my experiments.
So for reduce the amount of changes in this PR I propose do not change it in this PR.

Reply:
Fair enough.  Its good that the atomic implementation is good enough that reads alone are basically equivalent to just normal SLM reads.